packages/core-flows/src/price-list/steps/upsert-price-list-prices.ts

```diff
@@ -0,0 +1,195 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  AddPriceListPricesDTO,
+  CreatePriceListPriceDTO,
+  CreatePriceListPriceWorkflowDTO,
+  IPricingModuleService,
+  PriceListDTO,
+  PriceSetMoneyAmountDTO,
+  UpdatePriceListPriceDTO,
+  UpdatePriceListPriceWorkflowDTO,
+  UpdatePriceListPricesDTO,
+} from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  MedusaError,
+  arrayDifference,
+  buildPriceSetPricesForModule,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type WorkflowStepInput = {
+  price_list_id: string
+  prices?: (UpdatePriceListPriceWorkflowDTO | CreatePriceListPriceWorkflowDTO)[]
+}[]
+
+export const upsertPriceListPricesStepId = "upsert-price-list-prices"
+export const upsertPriceListPricesStep = createStep(
```

Comment:
question: Can we break this into separate steps for the workflow? 

We are not only upserting price list prices here, so the name is misleading. Also, it makes it less reusable. 

Wdyt/

Reply:
I did that initially 😄, it felt like an overkill since you have to run the same set of steps for both create and update. I'll give it another whirl. Maybe if i extract the variant links and price list checks I would be able to reuse for both.

Reply:
@olivermrbl made some changes, let me know wyt?

Reply:
Nice!